src/realm.h

```diff
@@ -251,6 +251,7 @@ typedef enum realm_logic_error_kind {
 typedef struct realm_error {
     realm_errno_e error;
     const char* message;
+    void* usercode_error;
```

Comment:
Can we add a comment that this is the SDK-provided opaque error value when `error == RLM_ERR_CALLBACK` with a callout to `realm_register_user_code_callback_error()`?